src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoListImpl.java

```diff
@@ -41,4 +41,12 @@ public ShoppingCart update(ShoppingCart shoppingCart) {
     public boolean delete(Long id) {
         return Storage.shoppingCarts.removeIf(cart -> cart.getId().equals(id));
     }
+
+    @Override
+    public ShoppingCart getByUserId(Long userId) {
+        return Storage.shoppingCarts.stream()
+                .filter(x -> x.getUserId().equals(userId))
```

Comment:
```suggestion
                .filter(s -> s.getUserId().equals(userId))
```
stream of shoppingCart, that's why  s instead of x

Reply:
ok